src/overlays/actors/ovl_Bg_Umajump/z_bg_umajump.c

```diff
@@ -27,34 +31,191 @@ const ActorInit Bg_Umajump_InitVars = {
+
+    if ((this->dyna.actor.params == 2)) {
+        if ((((play->sceneNum == SCENE_F01) && !(gSaveContext.save.weekEventReg[0x59] & 0x20)) &&
+             !CHECK_QUEST_ITEM(QUEST_SONG_EPONA)) &&
+            (this->dyna.actor.cutscene != -1)) {
+            this->actionFunc = BgUmajump_CheckDistance;
+            this->dyna.actor.update = func_8091A5A0;
+            this->dyna.actor.flags |= ACTOR_FLAG_10;
+            this->horse = SubS_FindActor(play, this->horse, ACTORCAT_BG, ACTOR_EN_HORSE);
+        } else {
+            this->dyna.actor.update = Actor_Noop;
+        }
+    } else {
+        this->objectIndex = Object_GetIndex(&play->objectCtx, OBJECT_UMAJUMP);
+
+        if (this->objectIndex < 0) {
+            Actor_MarkForDeath(&this->dyna.actor);
+        }
+
+        if ((this->dyna.actor.params == 3) && CHECK_QUEST_ITEM(QUEST_SONG_EPONA)) {
+            Actor_Spawn(&play->actorCtx, play, ACTOR_EN_KANBAN, this->dyna.actor.world.pos.x,
+                        this->dyna.actor.world.pos.y, this->dyna.actor.world.pos.z, this->dyna.actor.shape.rot.x,
+                        this->dyna.actor.shape.rot.y, this->dyna.actor.shape.rot.z, 0x3E);
+            Actor_MarkForDeath(&this->dyna.actor);
+        }
+    }
+}
+
+void BgUmajump_Destroy(Actor* thisx, PlayState* play) {
+    BgUmajump* this = THIS;
+
+    DynaPoly_DeleteBgActor(play, &play->colCtx.dyna, this->dyna.bgId);
+}
+
+void BgUmajump_Update(Actor* thisx, PlayState* play) {
+    BgUmajump* this = THIS;
+
+    if (Object_IsLoaded(&play->objectCtx, this->objectIndex)) {
+        this->dyna.actor.objBankIndex = this->objectIndex;
+        this->dyna.actor.draw = BgUmajump_Draw;
+
+        Actor_SetObjectDependency(play, &this->dyna.actor);
+
+        if (this->dyna.actor.params == 5) {
+            if (gSaveContext.save.weekEventReg[0x16] & 1) {
+                DynaPolyActor_LoadMesh(play, &this->dyna, &object_umajump_Colheader_001558);
+            }
+        } else if (this->dyna.actor.params == 6) {
+            if (!(gSaveContext.save.weekEventReg[0x16] & 1)) {
+                DynaPolyActor_LoadMesh(play, &this->dyna, &object_umajump_Colheader_001558);
+            }
+        } else if ((this->dyna.actor.params == 4) || (this->dyna.actor.params == 3)) {
+            DynaPolyActor_LoadMesh(play, &this->dyna, &object_umajump_Colheader_001558);
+        } else {
+            DynaPolyActor_LoadMesh(play, &this->dyna, &object_umajump_Colheader_001438);
+        }
+
+        if (this->dyna.actor.params == 1) {
+            this->objectIndex = 0;
+            this->actionFunc = func_80919F30;
+            this->dyna.actor.update = func_8091A5A0;
+        } else if (this->dyna.actor.params == 3) {
+            this->dyna.actor.update = func_8091A5A0;
+        } else if ((this->dyna.actor.params == 5) || (this->dyna.actor.params == 6)) {
+            if (this->dyna.actor.params == 5) {
+                if ((this->dyna.bgId == -1) && (gSaveContext.save.weekEventReg[0x16] & 1)) {
```

Comment:
```suggestion
                if ((this->dyna.bgId == BGACTOR_NEG_ONE) && (gSaveContext.save.weekEventReg[0x16] & 1)) {
```
Looks like we have a special define for -1 `bgId`. Could use this in the ohter places where it is appropriate please?